HCIDOCS-75: Add IPI installation workflow information to our document… (#94545)
openshift/openshift-docs: modules/ref_ipi-installation-overview.adoc

// Module included in the following assemblies:
//
// * installing/installing_bare_metal/ipi/ipi-install-prerequisites.adoc

:_mod-docs-content-type: REFERENCE

[id="installation-overview_{context}"]
= Installation overview

> The installation program supports interactive mode, but Red{nbsp}Hat recommends preparing an `install-config.yaml` file in advance, containing the provisioning details for all of the bare-metal hosts, and the relevant cluster details.

Reviewer: Per the SSG: [Avoid "recommends". Instead of "Red Hat recommends", direct users to take the recommended action](https://redhat-documentation.github.io/supplementary-style-guide/#recommend). Also, perhaps move the _in advance_ for better flow?
Maybe:

```suggestion
The installation program supports interactive mode. However, you can prepare an `install-config.yaml` file that contains the provisioning details for all of the bare-metal hosts, and the relevant cluster details, in advance.
```

> The installation program loads the `install-config.yaml` file, while the administrator generates the manifests and verifies all prerequisites.

Reviewer: Per ISG: Use _while_ only to refer to a period of time...
```suggestion
The installation program loads the `install-config.yaml` file and the administrator generates the manifests and verifies all prerequisites.
```

Author: Updated as suggested.

> The installation program enrolls all nodes in the cluster and starts the bootstrap VM. The installation program starts the metal platform components as `systemd` services, which have the following containers:

Author: A list is definitely a better solution. Thank you!!

* Ironic-dnsmasq: The DHCP server responsible for handing over the IP addresses to the provisioning interface of various nodes on the provisioning network. Ironic-dnsmasq is only enabled when you deploy an {product-title} cluster with a provisioning network.

> * Ironic-httpd: The HTTP server used to ship the images to the nodes.

Reviewer: I assume: 
```suggestion
* Ironic-httpd: The HTTP server that is used to ship the images to the nodes.
```
Otherwise, it could read as _used to_ means _previously_. As in the following mock example:
```suggestion
* Ironic-httpd: The HTTP server used to ship the images to the nodes. Now, it deletes them.
```

Author: Updated as suggested.

* Image-customization
* Ironic
* Ironic-inspector (available in {product-title} 4.16 and earlier)
* Ironic-ramdisk-logs

> * Extract-machine-os

Reviewer: I think that list items should be parallel. Perhaps, see the Red Hat Technical Writing Style Guide: [For each item in a list, start with either a complete sentence or a sentence fragment](https://stylepedia.net/style/#punc-in-lists).  I think this suggests if one item in the list is defined, they all should be.

* Provisioning-interface: The service that configures networking when the provisioning interface is enabled.

> * Metal3-baremetal-operator: The operator that provides a higher-level API for Ironic and replaces `terraform-provider-ironic` (available in {product-title} 4.16 and later)

Author: Updated as suggested.

> The nodes enter the validation phase, where each node moves to a “manageable” state after Ironic validates the credentials to access the Baseboard Management Controller (BMC).
> 
> Once in the “manageable” state, the “inspection” phase starts. The "inspection" phase ensures that the hardware meets the minimum requirements needed for a successful deployment of {product-title}.

Reviewer: I'm not sure the double-quotes are right here.  Are these states reflected in the GUI and should take back ticks?
Per ISG: [Do not use (once) as a conjunction to mean _after_ or _when_](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#o).
```suggestion
The nodes enter the validation phase, where each node moves to a _manageable_ state after Ironic validates the credentials to access the Baseboard Management Controller (BMC).

When the node is in the manageable state, the _inspection_ phase starts. The inspection phase ensures that the hardware meets the minimum requirements needed for a successful deployment of {product-title}.
```

> The `install-config.yaml` file details the provisioning network. The installation program on the bootstrap VM will try to push a live image by using PXE to every node with the Ironic Python Agent (IPA) loaded. When using virtual media, it connects directly to each node's BMC to virtually attach the image.

Author: I created this content off of a developer schematic, so it's possible the information wasn't correct.

> The `install-config.yaml` file details the provisioning network. The installation program on the bootstrap VM will try to push a live image by using PXE to every node with the Ironic Python Agent (IPA) loaded. When using virtual media, it connects directly to each node's BMC to virtually attach the image.

Author: Updated as suggested.

When using PXE boot, all nodes reboot to start the process:

* The `ironic-dnsmasq` service running on the bootstrap VM provides the IP address of the node and the TFTP boot server.
* The first-boot software loads the root file system over HTTP.
* The `ironic` service on the bootstrap VM receives the hardware information from each node.

The nodes then enter the “cleaning” state, where each node must clean all the disks before continuing with the configuration.

> Once the “cleaning” state finishes, the nodes enter the “available” state and the installation program moves the nodes to the “deploying” state.

Author: Updated as suggested.

> IPA runs the `coreos-installer` to install the {op-system-first} image on the disk defined by `rootDeviceHints` in the `install-config.yaml` file. The node boots by using {op-system}.

Reviewer: _the `coreos-installer`_ requires a noun. In a quick search, _command_, _utility_, _program_ all see common in the docs. Similar with `rootDeviceHints`, as it is in backticks.
```suggestion
IPA runs the `coreos-installer` command to install the {op-system-first} image on the disk defined by the `rootDeviceHints` parameter in the `install-config.yaml` file. The node boots by using {op-system}.
```

Author: Updated as suggested.

> Once the installation program configures the control plane nodes, it moves control from the bootstrap VM to the control plane nodes and deletes the bootstrap VM.

Reviewer: ```suggestion
After/when the installation program configures the control plane nodes, it moves control from the bootstrap VM to the control plane nodes and deletes the bootstrap VM.
```

> The baremetal-operator continues the deployment of the workers, storage, and infra nodes. Once the installation completes, the nodes move to the "active" state.

Author: Great catch! As it's a proper noun, it should actually be Bare Metal Operator (title case per BMO docs).

> The baremetal-operator continues the deployment of the workers, storage, and infra nodes. Once the installation completes, the nodes move to the "active" state.
> 
> Once the installation program finishes installation, you can proceed with postinstallation configuration and other Day 2 tasks.

Reviewer: ```suggestion
The baremetal-operator continues the deployment of the workers, storage, and infra nodes. 

After the installation completes, the nodes move to the active state. You can then proceed with postinstallation configuration and other Day 2 tasks.
```

Author: Updated as suggested.

The installation program performs the following tasks:

* Enrolls all nodes in the cluster
* Starts the bootstrap virtual machine (VM)
* Starts the metal platform components as `systemd` services, which have the following containers:

** Ironic-dnsmasq: The DHCP server responsible for handing over the IP addresses to the provisioning interface of various nodes on the provisioning network. Ironic-dnsmasq is only enabled when you deploy an {product-title} cluster with a provisioning network.
** Ironic-httpd: The HTTP server that is used to ship the images to the nodes.
** Image-customization
** Ironic
** Ironic-inspector (available in {product-title} 4.16 and earlier)
** Ironic-ramdisk-logs
** Extract-machine-os
** Provisioning-interface
** Metal3-baremetal-operator

The nodes enter the validation phase, where each node moves to a _manageable_ state after Ironic validates the credentials to access the Baseboard Management Controller (BMC).

When the node is in the manageable state, the _inspection_ phase starts. The inspection phase ensures that the hardware meets the minimum requirements needed for a successful deployment of {product-title}.

> The `install-config.yaml` file details the provisioning network. On the bootstrap VM, the installation program uses PXE to push a live image to every node with the Ironic Python Agent (IPA) loaded. When using virtual media, it connects directly to the BMC of each node to virtually attach the image.

Author: PXE stands for _Pre-Boot Execution Environment_. I've written out the acronym as suggested. Great catch!

When using PXE boot, all nodes reboot to start the process:

* The `ironic-dnsmasq` service running on the bootstrap VM provides the IP address of the node and the TFTP boot server.
* The first-boot software loads the root file system over HTTP.
* The `ironic` service on the bootstrap VM receives the hardware information from each node.

The nodes enter the cleaning state, where each node must clean all the disks before continuing with the configuration.

> Once the cleaning state finishes, the nodes enter the available state and the installation program moves the nodes to the deploying state.

Author: Updated as suggested.

IPA runs the `coreos-installer` command to install the {op-system-first} image on the disk defined by the `rootDeviceHints` parameter in the `install-config.yaml` file. The node boots by using {op-system}.

After the installation program configures the control plane nodes, it moves control from the bootstrap VM to the control plane nodes and deletes the bootstrap VM.

> The Bare Metal Operator continues the deployment of the workers, storage, and infra nodes.

Author: Updated as suggested.
